pkgs/build-support/fetchgitlab/default.nix

```diff
@@ -1,32 +1,62 @@
-{ fetchgit, fetchzip, lib }:
+{ lib, fetchgit, fetchzip }:
 
 # gitlab example
 { owner, repo, rev, protocol ? "https", domain ? "gitlab.com", name ? "source", group ? null
-, fetchSubmodules ? false, leaveDotGit ? false, deepClone ? false
+, fetchSubmodules ? false, leaveDotGit ? false
+, deepClone ? false, forceFetchGit ? false
+, sparseCheckout ? ""
```

Comment:
This is a boolean option?

Reply:
I don't think so. It looks rather like a string with directories/files in it:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchgit/tests.nix#L15
https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchgit/nix-prefetch-git#L51

Reply:
Then it should really be a list of strings rather than a multiline string.

Reply:
In principle I agree, but it would require a (breaking) change in [fetchgit](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchgit/default.nix#L18), [fetchFromGitHub](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchgithub/default.nix#L6), [documentation](https://nixos.org/manual/nixpkgs/stable/#fetchgit), and in some nixpkgs (although not so many and quite simple cases):

```console
$ git grep -l sparseCheckout
doc/builders/fetchers.chapter.md
pkgs/applications/misc/nerd-font-patcher/default.nix
pkgs/build-support/fetchgit/builder.sh
pkgs/build-support/fetchgit/default.nix
pkgs/build-support/fetchgit/nix-prefetch-git
pkgs/build-support/fetchgit/tests.nix
pkgs/build-support/fetchgithub/default.nix
pkgs/build-support/fetchgitlab/default.nix
pkgs/data/fonts/noto-fonts/default.nix
pkgs/data/icons/beauty-line-icon-theme/default.nix
pkgs/data/misc/colemak-dh/default.nix
pkgs/games/itch/default.nix
```

Reply:
As #200082 was finally merged, I changed `sparseCheckout` to a list (of strings).